parquet/src/main/java/org/apache/iceberg/parquet/ParquetSchemaUtil.java

```diff
@@ -112,4 +127,56 @@ public static MessageType addFallbackIds(MessageType fileSchema) {
 
     return builder.named(fileSchema.getName());
   }
+
+  public static MessageType addFallbackIds(MessageType fileSchema, NameMapping nameMapping) {
+    MessageTypeBuilder builder = org.apache.parquet.schema.Types.buildMessage();
```

Comment:
This is should recursively assign IDs using a visitor, not just assign to the top-level.

Reply:
I think we need a test suite for this assignment, too.

Reply:
Hmm. My bad. Will update in the next commit.